Cannot complete build with ENABLE_WEB_SERVICE OFF

Yuzu build fails when I try to run cmake with following options, I used mainline repo [commit tree](https://github.com/yuzu-emu/yuzu-mainline/tree/b241a35ed7079bcb3980b424aa99787ef2d5a45e)

```
	cmake .. \
	  -DCMAKE_INSTALL_PREFIX=/usr \
	  -DCMAKE_BUILD_TYPE=Release \
	  -DYUZU_USE_BUNDLED_UNICORN=ON \
	  -DYUZU_ENABLE_COMPATIBILITY_REPORTING=OFF \
	  -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=OFF \
          -DENABLE_WEB_SERVICE=OFF \
          -DYUZU_ENABLE_BOXCAT=OFF \
	  -DUSE_DISCORD_PRESENCE=OFF
```

basically when `ENABLE_WEB_SERVICE=OFF` it fails to include `json.hpp` required for `reporter.cpp`

and gives following error
```
/tmp/yuzu/src/yuzu-mainline/src/core/reporter.cpp:11:10: fatal error: json.hpp: No such file or directory
   11 | #include <json.hpp>
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,10 @@ option(ENABLE_CUBEB "Enables the cubeb audio backend" ON)
 
 option(USE_DISCORD_PRESENCE "Enables Discord Rich Presence" OFF)
 
+if (NOT ENABLE_WEB_SERVICE)
+    set(YUZU_ENABLE_BOXCAT OFF)
+endif()
+
 # Default to a Release build
 get_property(IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
 if (NOT IS_MULTI_CONFIG AND NOT CMAKE_BUILD_TYPE)
@@ -239,7 +243,7 @@ if(ENABLE_QT)
     if (YUZU_USE_QT_WEB_ENGINE)
         find_package(Qt5 COMPONENTS WebEngineCore WebEngineWidgets)
     endif()
-	
+
     if (ENABLE_QT_TRANSLATION)
         find_package(Qt5 REQUIRED COMPONENTS LinguistTools ${QT_PREFIX_HINT})
     endif()
@@ -322,7 +326,7 @@ if (CONAN_REQUIRED_LIBS)
             list(APPEND Boost_LIBRARIES Boost::context)
         endif()
     endif()
-    
+
     # Due to issues with variable scopes in functions, we need to also find_package(qt5) outside of the function
     if(ENABLE_QT)
         list(APPEND CMAKE_MODULE_PATH "${CONAN_QT_ROOT_RELEASE}")

